src/main/java/uk/gov/justice/laa/portal/landingpage/controller/UserController.java

```diff
@@ -103,30 +103,52 @@ public String displayAllUsers(
             @RequestParam(name = "showFirmAdmins", required = false) boolean showFirmAdmins,
             Model model, HttpSession session, Authentication authentication) {
 
-        PaginatedUsers paginatedUsers;
+        PaginatedUsers paginatedUsers = new PaginatedUsers();
         EntraUser entraUser = loginService.getCurrentEntraUser(authentication);
         boolean internal = userService.isInternal(entraUser.getId());
         boolean canSeeAllUsers = accessControlService.authenticatedUserHasPermission(Permission.VIEW_INTERNAL_USER)
                 && accessControlService.authenticatedUserHasPermission(Permission.VIEW_EXTERNAL_USER);
         List<Permission> permissions = new ArrayList<>();
+        List<UserType> userTypesToFilter;
+
         if (showFirmAdmins) {
-            permissions.add(Permission.CREATE_EXTERNAL_USER);
+            // When showFirmAdmins is true,
+            // filter to only show EXTERNAL_SINGLE_FIRM_ADMIN users
+            userTypesToFilter = List.of(UserType.EXTERNAL_SINGLE_FIRM_ADMIN);
+        } else {
+            // Default to all external user types if showFirmAdmins is false
+            userTypesToFilter = UserType.EXTERNAL_TYPES;
         }
+
         if (canSeeAllUsers) {
-            paginatedUsers = userService.getPageOfUsersByNameOrEmailAndPermissionsAndFirm(search, permissions, null, null,
-                    page, size, sort, direction);
+            // Combine userTypesToFilter with INTERNAL user type
+            List<UserType> allUserTypes = new ArrayList<>(userTypesToFilter);
+            if (!showFirmAdmins) {
+                allUserTypes.add(UserType.INTERNAL);
+            }
+            paginatedUsers = userService.getPageOfUsersByNameOrEmailAndPermissionsAndFirm(
+                    search, permissions, null, allUserTypes, page, size, sort, direction);
         } else if (accessControlService.authenticatedUserHasPermission(Permission.VIEW_INTERNAL_USER)) {
-            paginatedUsers = userService.getPageOfUsersByNameOrEmailAndPermissionsAndFirm(search, permissions, null, List.of(UserType.INTERNAL),
-                    page, size, sort, direction);
+            permissions.add(Permission.VIEW_INTERNAL_USER);
```

Comment:
this VIEW_INTERNAL_USER permission was removed at this PR? 
https://github.com/ministryofjustice/laa-landing-page/pull/353/files